### DISCUSSION TITLE
When creating a db hook, I can't read the inserted data from the e.model object

### DISCUSSION THREAD

**Discussion by jmaris** [pocketbase/pocketbase#5573] [Q&A] [answered]
Hey,

I'm losing my mind here, and beginning to feel that I _may_ be going insane. I'm trying to trigger an action upon insertion of a record into the database. In short, when a record is inserted into the database, I need to get consultation_id (not the pocketbase id) from that record and use it for further processing.

```
onModelAfterCreate((e) => {
  console.log("%j",e.model);
  console.log(e.model.consultation_id);
// other processing goes here
}, "consultations");
```

But when I run this, the output in console is as follows:
```
2024/09/26 13:43:48 {"archived":false,"collectionId":"ca0lug7faxdeh2j","collectionName":"consultations","consultation_description":"","consultation_events":[],"consultation_id":"14415","consultation_name":"","created":"2024-09-26 11:43:48.029Z","id":"4wt3kldyiukzn72","topic":[],"updated":"2024-09-26 11:43:48.029Z","url":""}
2024/09/26 13:43:48 

```
As you can see, the data I wanted has been added to the database, and is part of e.model, but I when i try to use it, It returns nothing.

Is there something I really don't understand about how this works? I'm at a complete loss x(

**Comment by ganigeorgiev** [comment, marked as answer]
At the moment the dynamic record fields can be accessed or modified only through the `get/set` methods, ex. `e.model.get("consultation_id)` (for all record helpers you could check https://pocketbase.io/docs/js-records/#getset-record-fields).

**Reply by jmaris** [reply]
Thanks! I'm sorry I missed that!

**Reply by jmaris** [reply]
Out of curiosity, I noted you said "at the moment", does this mean there are plans to refactor this?

**Reply by ganigeorgiev** [reply]
It is just how in general the JSVM works at the moment, and there is no easy way to fix it with refactoring from PocketBase.

There are plans to _experiment_ with attempting to define a global JS [`Proxy`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy) that under the hood will resolve to the designated get/set record methods but remains a low priority for now (_I'm not even sure whether this will be possible at all and it just something that exists in my to do to check_).

The reason why we have to use get/set methods is because that is how you access the dynamic record fields in Go. And because the JSVM operates directly with the Go structs, methods, functions, etc. it mimics the same apis via reflection and the JS interpreter (goja) doesn't allow much flexibility in this regard (this may change if eventually https://github.com/dop251/goja/issues/426 gets implemented).
